src/core/agent/services/signifyNotificationService.ts

```diff
@@ -168,6 +169,25 @@ class SignifyNotificationService extends AgentService {
     if (notif.r) {
       return;
     }
+    if (notif.a.r === NotificationRoute.ExnIpexApply) {
+      const existingNotficationRecord = await this.notificationStorage.findById(
+        notif.i
+      );
+      if (!existingNotficationRecord) {
```

Comment:
We should be checking if an IPEX message record exists with a certain ID. Based on my other comment the ID will be the SAID of the ipex exn message.

(doesn't really make sense to check if the notification exists during to certain failures, and the notification gets deleted etc)